Market Tools/macd_strategy.py

```diff
@@ -0,0 +1,147 @@
+# Import the required libraries
+import numpy as np
+import pandas as pd
+import hvplot.pandas
+from pathlib import Path
+from datetime import date
+from pandas.core.frame import DataFrame
+import yfinance as yf
+import os
+import pricing
+
+def analyze_macd(signals_df:DataFrame):
+  # Calculate the MACD and Signal Line Indicators
+  # Calculate the short term exponential moving average (EMA)
+  signals_df["short_ema"] = signals_df.Close.ewm(span=12, adjust=False).mean()
+
+  # Calculate the long term exponential moving average (EMA)
+  signals_df["long_ema"] = signals_df.Close.ewm(span=26, adjust=False).mean()
+
+  # Calculate the MACD Line
+  signals_df["macd"] = signals_df["short_ema"] - signals_df["long_ema"]
+
+  # Calculate the Signal Line
+  signals_df["signal line"] = signals_df["macd"].ewm(span=9, adjust=False).mean()
+
+  # Create a column to hold the trading signal
+  signals_df["Signal"] = 0.0
+  
+  # Generate the trading signal 0 or 1,
+  # where 1 is the MACD greater than the signal line
+  # and 0 is when the condition is not met
+  signals_df["Signal"] = np.where(
+      signals_df["macd"] > signals_df["signal line"], 1.0, 0.0
+  )
+
+  # Calculate the points in time when the Signal value changes
+  # Identify trade entry (1) and exit (-1) points
+  signals_df["Entry/Exit"] = signals_df["Signal"].diff()
+
+  # Review the DataFrame
+  signals_df.loc["2015-12-03":"2015-12-13"]
```

Comment:
[Needs to be fixed this PR] This line of code can cause the program to fail a loaded ticker did not exist on this date in 2015.  An alternative approach would be to check for the amount of rows in the DataFrame and if greater than 0 (in case there are no rows!) use the iloc method to display rows by index number instead of a specific date.  In the grand scheme of things though, is this line of code just for the developer's benefit?  If it is, then consider wrapping such debug statements in a conditional statement like the one below, where DEBUG_MODE is some app-level configuration constant you may define as True or False. 

```
if DEBUG_MODE:
    ....
```

Reply:
Resolved issue by commenting out code that included 2015.